aea/contracts/base.py

```diff
@@ -191,3 +198,24 @@ def get_state(
         :return: the tx
         """
         raise NotImplementedError
+
+
+def _try_to_register_contract(configuration: ContractConfig):
+    """Register a contract to the registry."""
+    if str(configuration.public_id) in contract_registry.specs:
+        logger.warning(
+            f"Skipping registration of contract {configuration.public_id} since already registered."
+        )
+        return
+    logger.debug(f"Registering contract {configuration.public_id}")  # pragma: nocover
+    try:  # pragma: nocover
+        contract_registry.register(
+            id_=str(configuration.public_id),
+            entry_point=f"{configuration.prefix_import_path}.contract:{configuration.class_name}",
+            class_kwargs={"contract_interface": configuration.contract_interfaces},
```

Comment:
This parameter will make the registry to populate the contract class with the attributes specified in the dictionary.
However, this cannot be overwritten by `registry.make`, so the class attributes will be set in the first registration and never changed (in the same process). This might lead to unexpected behaviours of the framework, and related to issue #1438

Reply:
That's a feature rather than a bug from my perspective